distributionviewer/core/static/js/app/components/views/chart-hover.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import * as d3Array from 'd3-array';
+import { select, mouse } from 'd3-selection';
+
+
+export default class extends React.Component {
+  componentDidMount() {
+    let hoverElm = select(this.refs.rect);
+    this.focusElm = select(`.chart-${this.props.id} .focus`);
+    this.bisector = d3Array.bisector(d => d.x).left;
+
+    // Terrible hack to bind an event in a way d3 prefers.
+    // Normally this would be in the container and we'd pass it the event.
+    hoverElm.on('mousemove', () => {
+      this._handleMouseMove();
+    });
+  }
+  _handleMouseMove() {
+    let props = this.props;
+    let x0 = props.xScale.invert(mouse(this.refs.rect)[0]);
+    let i = this.bisector(props.data, x0, 1);
+    let d0 = props.data[i - 1];
+    let d1 = props.data[i];
+    let d = x0 - d0.x > d1.x - x0 ? d1 : d0;
```

Comment:
After hovering over charts for a while, I get an error in console about `d1` being undefined. (Chrome instead says that `x` is undefined.)